### PR TITLE
Feature/ab#48 events backend

### PR DESCRIPTION
**Pull request by vyruz1986**
See note in last commit:
In order to be able to use the `HasFlag` functionality of `Enum`, the values of the `AgeGroup` enumerations had to be changed to base 2 increments, might need further work in `CareUser` related stuff

### CODE REVIEW

File: Singer.API/DTOs/CreateEventLocationDTO.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+namespace Singer.DTOs
+{
+   public class CreateEventLocationDTO
+   {
+      public string Name { get; set; }
+      public string Adress { get; set; }
```
Comment (BerendWouters) [inline Singer.API/DTOs/CreateEventLocationDTO.cs]: Typo: Ad**d**ress

File: Singer.API/DTOs/EventDTO.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+   public class EventDTO : IIdentifiable
+   {
+      public Guid Id { get; set; }
+      public string Title { get; set; }
+      public string Description { get; set; }
+      public List<AgeGroup> AllowedAgeGroups { get; set; }
+      public EventLocationDTO Location { get; set; }
+      public int MaxRegistrants { get; set; }
+      public decimal Cost { get; set; }
+      public DateTimeOffset StartDate { get; set; }
```
Comment (BerendWouters) [inline Singer.API/DTOs/EventDTO.cs]: Why using DateTimeOffset, and not just DateTime?

File: Singer.API/DTOs/EventLocationDTO.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+using System;
+using Singer.Helpers;
+
+namespace Singer.DTOs
+{
+   public class EventLocationDTO : IIdentifiable
+   {
+      public Guid Id { get; set; }
+      public string Name { get; set; }
+      public string Adress { get; set; }
```
Comment (BerendWouters) [inline Singer.API/DTOs/EventLocationDTO.cs]: Same typo

File: Singer.API/Models/EventLocation.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+using System;
+using Singer.Helpers;
+
+namespace Singer.Models
+{
+   public class EventLocation : IIdentifiable
+   {
+      public Guid Id { get; set; }
+      public string Name { get; set; }
+      public string Adress { get; set; }
```
Comment (BerendWouters) [inline Singer.API/Models/EventLocation.cs]: Typo :)

File: Singer.API/Models/Event.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      public Guid Id { get; set; }
+      public string Title { get; set; }
+      public string Description { get; set; }
+      public AgeGroup AllowedAgeGroups { get; set; }
+      [ForeignKey(nameof(Location))]
+      public Guid LocationId { get; set; }
+      public EventLocation Location { get; set; }
+      public int MaxRegistrants { get; set; }
+      public decimal Cost { get; set; }
+      public DateTimeOffset StartDate { get; set; }
```
Comment (BerendWouters) [inline Singer.API/Models/Event.cs]: Again, why DateTimeOffset?

File: Singer.API/DTOs/CreateEventLocationDTO.cs (hunk 1 above)
Comment (vyruz1986) [reply Singer.API/DTOs/CreateEventLocationDTO.cs]: Fixed

File: Singer.API/DTOs/EventLocationDTO.cs (hunk 3 above)
Comment (vyruz1986) [reply Singer.API/DTOs/EventLocationDTO.cs]: Fixed

File: Singer.API/Models/Event.cs (hunk 5 above)
Comment (vyruz1986) [reply Singer.API/Models/Event.cs]: Fixed

File: Singer.API/Models/EventLocation.cs (hunk 4 above)
Comment (vyruz1986) [reply Singer.API/Models/EventLocation.cs]: Fixed

**Review by BerendWouters — APPROVED** [review]

**Pull request merged**